src/cclib/parser/orcaparser.py

```diff
@@ -643,23 +643,144 @@ def extract(self, inputfile, line):
+            if (line[16:24] == 'COMBINED' and not 'ROCIS' in line) or \
+	       (line[ 6:14] == 'COMBINED' and 'origin adjusted' in line):
+                def energy_intensity(line):
+                    """ TDDFT with DoQuad == True
+------------------------------------------------------------------------------------------------------
+                COMBINED ELECTRIC DIPOLE + MAGNETIC DIPOLE + ELECTRIC QUADRUPOLE SPECTRUM
+------------------------------------------------------------------------------------------------------
+State   Energy Wavelength    D2        m2        Q2         D2+m2+Q2       D2/TOT    m2/TOT    Q2/TOT
+        (cm-1)   (nm)                (*1e6)    (*1e6)
+------------------------------------------------------------------------------------------------------
+   1 61784150.6      0.2   0.00000   0.00000   3.23572   0.00000323571519   0.00000   0.00000   1.00000
+"""
+                    state, energy, wavelength, d2, m2, q2, intensity, d2_contrib, m2_contrib, q2_contrib = line.split()
+                    return energy, intensity
+
+            elif line[36:44] == 'COMBINED':
+                def energy_intensity(line):
+                    """ TDDFT with doQuad == True (Origin Independent Length Representation)
+-------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
+                                    COMBINED ELECTRIC DIPOLE + MAGNETIC DIPOLE + ELECTRIC QUADRUPOLE SPECTRUM (Origin Independent, Length Representation)
+-------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
+State  Energy   Wavelength       D2            m2              Q2               DM             DO               D2+m2+Q2+DM+DO          D2/TOT          m2/TOT          Q2/TOT         DM/TOT          DO/TOT
+       (cm-1)      (nm)                      (*1e6)          (*1e6)           (*1e6)         (*1e6)
+-------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
+   1 61784150.6      0.2      0.00000         0.00000         3.23572         0.00000         0.00000         0.00000323571519         0.00000         0.00000         1.00000         0.00000          0.00000
+   2 61793079.3      0.2      0.00000         0.00000         2.85949         0.00000        -0.00000         0.00000285948800         0.00000         0.00000         1.00000         0.00000         -0.00000
+"""
+                    try:
+                        state, energy, wavelength, d2, m2, q2, dm, do, intensity, d2_contrib, m2_contrib, q2_contrib, dm_contrib, do_contrib = line.split()
```

Comment:
This was the most compact yet readable format that I could think of. I'll see what I can do with multiple lines.